File: 3451-string-compression-iii/string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:

        i = 0
        res =""

        while i < len(word):
            current = word[i]
            count = 0

            while i < len(word) and current == word[i]:
                i+=1
                count+=1
                if count == 9:
                    res += str(count)+current
                    count = 0
            if count >=1:
                res += str(count)+current
        return res        

File: 3451-string-compression-iii/string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        
        String res ="";
        int i = 0;

        while(i<word.length())
        {
            char current = word.charAt(i);
            int count = 0;

            while(i<word.length() && word.charAt(i)==current)
            {
                i++;
                count++;
                if(count==9)
                {
                    res += count+String.valueOf(current);
                    count = 0;
                }
            }
            
            if(count>=1)
            res += count+String.valueOf(current);


        }

        return res;
    }
}